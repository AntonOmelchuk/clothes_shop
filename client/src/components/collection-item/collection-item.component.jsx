import React from 'react';
import {connect} from 'react-redux';

import {addItem} from '../../redux/cart/cart.action';

import {
  AddButton,
  CollectionFooter,
  CollectionItemContainer,
  ImageContainer,
  ItemName,
  ItemPrice,
} from './collection-item.style';

export const CollectionItem = ({item, addItem}) => {
  const {name, price, imageUrl} = item;

  return (
    <CollectionItemContainer>
      <ImageContainer className='image' imageUrl={imageUrl} />
      <CollectionFooter>
        <ItemName>{name}</ItemName>
        <ItemPrice>${price}</ItemPrice>
      </CollectionFooter>
      <AddButton inverted={true} onClick={() => addItem(item)}>
        add to cart
      </AddButton>
    </CollectionItemContainer>
  );
};

export default connect(null, {addItem})(CollectionItem);
