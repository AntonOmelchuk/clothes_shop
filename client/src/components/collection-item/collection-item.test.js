import React from 'react';
import {shallow, render} from 'enzyme';
import {CollectionItem} from './collection-item.component';

describe('CollectionItem component', () => {
  let wrapper;
  let mockProps;
  let mockAddItem;
  const imageUrl = 'www.test.com';
  const mockName = 'test name';
  const mockPrice = 21;

  beforeEach(() => {
    mockAddItem = jest.fn();

    mockProps = {
      item: {
        imageUrl: imageUrl,
        price: mockPrice,
        name: mockName,
      },
      addItem: mockAddItem,
    };

    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  it('renders CollectionItem component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
