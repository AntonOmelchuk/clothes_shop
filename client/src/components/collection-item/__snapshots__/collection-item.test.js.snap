// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component renders CollectionItem component 1`] = `ShallowWrapper {}`;
